third_party/nvfuser/csrc/python_frontend/python_bindings.cpp

```diff
@@ -140,18 +140,32 @@ void initNvFuserPythonBindings(PyObject* module) {
           })
       .def(
           "add_output",
-          [](nvfuser::FusionDefinition& self, nvfuser::Scalar output) {
+          [](nvfuser::FusionDefinition& self, nvfuser::Scalar output, c10::optional<nvfuser::Scalar> alias_input = c10::nullopt) {
             FUSER_PERF_SCOPE("FusionDefinition.add_output (scalar)");
-            self.defineRecord(new nvfuser::OutputRecord<Nvf::Val>(
-                {self.recordingState(output())}));
-          })
+            if (alias_input.has_value()) {
+              self.defineRecord(new nvfuser::OutputRecord<Nvf::Val>(
+                  {self.recordingState(output()), self.recordingState(alias_input.value()())}));
+            } else {
+              self.defineRecord(new nvfuser::OutputRecord<Nvf::Val>(
+                  {self.recordingState(output())}));
```

Comment:
nitpick: instead of variadic record, maybe we can just have default for alias entries? Something like a `None`.

Maybe it's too late already, I'm not totally sure if there's already pattern in the code base using this :)

Reply:
I got rid of the Scalar aliasing.